Fix: When clicking "Download Handout", the handout (image) overlays the video area, and the download does not start automatically.

haftamuk opened the pull request:
<!--

Note: Please refer to the Support Development Guidelines on the wiki page to consider backporting to active releases:
https://openedx.atlassian.net/wiki/spaces/COMM/pages/4248436737/Support+Guidelines+for+active+releases

Please give your pull request a short but descriptive title.
Use conventional commits to separate and summarize commits logically:
https://open-edx-proposals.readthedocs.io/en/latest/oep-0051-bp-conventional-commits.html

Use this template as a guide. Omit sections that don't apply.
You may link to information rather than copy it, but only if the link is publicly readable.
If the linked information must be private (because it contains secrets), clearly label the link as private.

-->

## Description
This pull request will enable learners to be able to click "Download Handout" link available in a video component and initiate a direct download on image attachments(image handouts). The "download" attribute is added in the anchor element to perform a direct download.


Learners exploring courses with video component and an image handout attached to it will be able to click the "Download Handout" link to initiate a direct download.

## Supporting information

- [Issue Link](https://github.com/openedx/edx-platform/issues/37081)

- [a: The Anchor element](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/a#download)

## Testing instructions

- [You can follow instructions available here to reproduce the issue](https://github.com/openedx/edx-platform/issues/37081)

## Deadline

None

## Other information

None

haftamuk committed:
fix: Download Handout link in a video component overlaying over the video

 After applying a "download" property on html5 anchor tag, Download Handout
link in a video component initiates a direct download instead of overlaying
over the video.

Fixed issue: https://github.com/openedx/edx-platform/issues/37081

Cup0fCoffee commented:
@haftamuk The changes look good, I verified that they do fix the issue.

Can you please:
* [Rebase your branch](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/using-git.html#rebasing-your-branch) onto `openedx/master` and force push?
* Fix the PR description - some of the formatting looks off and the link to the testing instructions contains wrong text.
* Fix the [format of your commit message](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/landing-your-work.html#commit-messages-and-rebasing)?

openedx-webhooks commented:
Thanks for the pull request, @haftamuk!


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

e0d commented:
@haftamuk it looks like there are some branch conflicts that need to be resolved here.  Can you have a look?

Cup0fCoffee commented:
I think the conflicts will be resolved once the changes are rebased onto `master` branch, because currently the branch is based on `release/teak` (because the local tutor is more stable with release branch than with master).

haftamuk commented:
@e0d I have rebased the changes on top of master and have resolved the conflicts. Is there anything else I can do to get this reviewed?

e0d commented:
@haftamuk Your branch is behind the base.  I've pulled in changes from master as a merge commit which will update your branch and cause the tests to be re-run.

haftamuk commented:
> @haftamuk Your branch is behind the base. I've pulled in changes from master as a merge commit which will update your branch and cause the tests to be re-run.

Ok. Great thank you.

mphilbrick211 commented:
Hi @haftamuk! Is this pull request still in progress?

haftamuk commented:
@mphilbrick211 Can you point me in to the next directions, I am thinking it is complete. Kindly let me know why is it not merged yet?

Cup0fCoffee commented:
@haftamuk can you please rebase onto the target branch and squash your commits, so that your git history is clean and passes the linter checks? It might take a while before an upstream reviewer can take a look at your PR. Making sure your PR can pass all checks will reduce the back and forth required, once someone finds the time to look at the PR.

Also, can you please link the issue id in PR description? **Edit**: I see that you linked it in the commit (which is good!), just linking it in the PR description as well makes it easier to find and navigate between the PR and the issue.

haftamuk commented:
> @haftamuk can you please rebase onto the target branch and squash your commits, so that your git history is clean and passes the linter checks? It might take a while before an upstream reviewer can take a look at your PR. Making sure your PR can pass all checks will reduce the back and forth required, once someone finds the time to look at the PR.
> 
> Also, can you please link the issue id in PR description? **Edit**: I see that you linked it in the commit (which is good!), just linking it in the PR description as well makes it easier to find and navigate between the PR and the issue.

Thank you for the followup
I have rebased it on master and squashed my commits. The PR shows 8 commits from other contributors(I am not sure how to deal with that), but I only had one original(meaningful) commit and the second amend commit. I have squashed this two, but the other commits remain. Let me know If I should do cleaning on those!

I have edited the PR description to contain the Issue link.
Thank you.

Cup0fCoffee commented:
@haftamuk Yeah, you should remove those other commits from your branch. I'm not sure how they ended up in your branch, so not sure what's the best way to remove it.

haftamuk commented:
> @haftamuk Yeah, you should remove those other commits from your branch. I'm not sure how they ended up in your branch, so not sure what's the best way to remove it.

It is clean now! The  change in this PR is really simple.

Cup0fCoffee commented:
@haftamuk Awesome, thank you!

e0d commented:
@haftamuk 
Your branch is behind the base.  I've pulled in changes from master as a merge commit which will update your branch and cause the tests to be re-run.

feanil commented:
@haftamuk this looks good, cane you rebase the code and fix the conflict? Then I can merge it.

haftamuk commented:
> @haftamuk this looks good, cane you rebase the code and fix the conflict? Then I can merge it.

@feanil I did resolved the conflict and rebased the PR
Kindly have a look

Thank you
